node/src/neighborhood/mod.rs

```diff
@@ -2087,17 +2049,75 @@ impl Neighborhood {
     }
 }
 
-pub fn regenerate_signed_gossip(
-    inner: &NodeRecordInner_0v1,
-    cryptde: &dyn CryptDE, // Must be the correct CryptDE for the Node from which inner came: used for signing
-) -> (PlainData, CryptData) {
-    let signed_gossip =
-        PlainData::from(serde_cbor::ser::to_vec(&inner).expect("Serialization failed"));
-    let signature = match cryptde.sign(&signed_gossip) {
-        Ok(sig) => sig,
-        Err(e) => unimplemented!("TODO: Signing error: {:?}", e),
-    };
-    (signed_gossip, signature)
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ExitLocationsRoutes<'a> {
+    routes: Vec<(Vec<&'a PublicKey>, i64)>,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum FallbackPreference {
+    Nothing,
+    ExitCountryWithFallback,
+    ExitCountryNoFallback,
+}
+
+//TODO create big comment about all members and its utilization
```

Comment:
Tady mas jeste dve TODO (u toho fieldu locations_opt je druhe). Byl bych vazne rad, kdybychom to tady vysvetlene meli.